Sources/NIOHTTP2/HTTP2FrameParser.swift

```diff
@@ -0,0 +1,835 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHPACK
+
+/// Ingests HTTP/2 data and produces frames. You feed data in, and sometimes you'll get a complete frame out.
+struct HTTP2FrameDecoder {
+    private var headerDecoder: HPACKDecoder
+    
+    private struct IgnoredFrame : Error {}
+    
+    private enum ParserState {
+        /// We are not in the middle of parsing any frames.
+        case idle
+        
+        /// We are accumulating payload bytes for a single frame.
+        case accumulatingData(FrameHeader, ByteBuffer)
```

Comment:
I prefer to leave these out when it's fairly clear what they are. The others have labels because it's not immediately obvious what the difference is between the various buffers or integer types they contain.